warning: comparison of integers of different signs

Adding -Wextra yields:

utf8proc.c:439:28: warning: comparison of integers of different signs: 'utf8proc_ssize_t' (aka 'long') and 'unsigned long' [-Wsign-compare]
      if (wpos < 0 || wpos > SSIZE_MAX/sizeof(utf8proc_int32_t)/2)
                      ~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This is an issue if a project embeds utf8proc and it's compiled with -Wextra -Werror.

--- a/utf8proc.c
+++ b/utf8proc.c
@@ -436,7 +436,8 @@ UTF8PROC_DLLEXPORT utf8proc_ssize_t utf8proc_decompose(
       if (decomp_result < 0) return decomp_result;
       wpos += decomp_result;
       /* prohibiting integer overflows due to too long strings: */
-      if (wpos < 0 || wpos > SSIZE_MAX/sizeof(utf8proc_int32_t)/2)
+      if (wpos < 0 ||
+          wpos > (utf8proc_ssize_t)(SSIZE_MAX/sizeof(utf8proc_int32_t)/2))
         return UTF8PROC_ERROR_OVERFLOW;
     }
   }

